### ISSUE TITLE
AWS-CLI v2.1.28 do not support UTF8 caracters in YAML file anymore

### ISSUE THREAD

**Issue by shavounet** [aws/aws-cli#5973]
Confirm by changing [ ] to [x] below to ensure that it's a bug:
- [x] I've gone though the [User Guide](https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-welcome.html) and the [API reference](https://docs.aws.amazon.com/cli/latest/reference/)
- [x] I've searched for [previous similar issues](https://github.com/aws/aws-cli/issues) and didn't find any solution
  
**Describe the bug**
In my CI, a `aws cloudformation deploy` job with an ECS Yaml template that was ok yesterday is failing now with the exact same git commit, and same environment variables.

Error message is `'ascii' codec can't decode byte 0xc3 in position 9868: ordinal not in range(128)`.

In my ECS file contains this (1st line only)
```
        Cooldown: 60 # Délai d'attente avant une nouvelle opération d'Auto Scaling
                        ^ character 9868
```

**SDK version number**
- v2.1.28

**Platform/OS/Hardware/Device**
- Docker image `amazon/aws-cli:latest` (at the time of the writing : v2.1.28)

**To Reproduce (observed behavior)**
- Create some ECS template with a `é` character in a comment (or any other non ascii comment ?)
- Trigger the deploy procedure with `docker-compose run aws cloudformation deploy --template ${AWS_ECS_FILE} ...` (supposing you have a docker compose file with service `aws` using image `amazon/aws-cli:2.1.28` )

**Expected behavior**
CLI output in v2.1.27 with the _exact same file and environment_ is 
```
Waiting for changeset to be created..
Waiting for stack create/update to complete
Successfully created/updated stack - ECSP
```

**Logs/output**
Not available due to CI context.

**Additional context**
Nothing else to add

**Comment by kdaily** [comment]
Thanks for the report, @shavounet.

Can you share your `--debug` log? I'll look into this.

**Comment by kdaily** [comment]
There were changes both to the bundler (PyInstaller) and an update to use the Python 3.8 runtime, so we'll need to look more deeply into this. Can you use the 2.1.17 container image (`amazon/aws-cli:2.1.27`)?

**Comment by kdaily** [comment]
Stack trace running from container image `amazon/aws-cli:2.1.28`:

```
2021-02-23 21:45:38,170 - MainThread - botocore.hooks - DEBUG - Event creating-client-class.cloudformation: calling handler <function add_generate_presigned_url at 0x7f49d7715ee0>
2021-02-23 21:45:38,173 - MainThread - botocore.endpoint - DEBUG - Setting cloudformation timeout as (60, 60)
2021-02-23 21:45:38,178 - MainThread - awscli.clidriver - DEBUG - Exception caught in main()
Traceback (most recent call last):
  File "awscli/clidriver.py", line 457, in main
  File "awscli/clidriver.py", line 586, in __call__
  File "awscli/customizations/commands.py", line 191, in __call__
  File "awscli/customizations/cloudformation/deploy.py", line 317, in _run_main
  File "/usr/local/cpython-exe-builder/lib/python3.8/encodings/ascii.py", line 26, in decode
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 55: ordinal not in range(128)

'ascii' codec can't decode byte 0xc3 in position 55: ordinal not in range(128)
```

**Comment by kdaily** [comment]
Looks like this is only affecting the container image, we'll investigate more to determine what's the underlying change. However, supplying `-e AWS_CLI_FILE_ENCODING='utf-8'` to your Docker command will address this immediately and allow you to use the `latest` container image.

**Comment by kdaily** [comment]
Hi @shavounet, this should be addressed in version 2.1.29 available from the container image:

PR: https://github.com/aws/aws-cli/pull/5980
Changelog entry: https://github.com/aws/aws-cli/blob/v2/CHANGELOG.rst#2129

Thanks again for reporting!
